django_socio_grpc/protobuf/proto_classes.py

```diff
@@ -179,22 +182,17 @@ def from_typing(
 
         if proto_type := TYPING_TO_PROTO_TYPES.get(get_origin(field_typing) or field_typing):
             field_type = proto_type
-
-        if isinstance(field_typing, ProtoEnum):
+        elif isinstance(field_typing, ProtoEnum):
             field_type = field_typing
         elif isinstance(field_typing, type) and issubclass(field_typing, Enum):
             field_type = ProtoEnum(field_typing)
-
-        if not field_type:
+        elif is_dataclass(field_typing):
             # Handle nested dataclasses
-            if hasattr(field_typing, "__dataclass_fields__"):
-                field_type = ProtoMessage.from_dataclass(
-                    field_typing, name=field_typing.__name__
-                )
-            else:
-                raise ProtoRegistrationError(
-                    f"Unknown field type `{field_typing}` for field `{field_name}`"
-                )
+            field_type = ProtoMessage.from_dataclass(field_typing, name=field_typing.__name__)
+        else:
```

Comment:
I would have put a separate if not field_type

To avoid having a None field_type passing because one of the previous condition set it to None